### PR TITLE
feature/jenkins 36586 - extensions table actions

### PR DESCRIPTION
**Pull request by cliffmeyers**
**Decription**
This adds some extension points to various screens in the UI.

The change is completely inert until someone actually makes use of the extension points. I did have one XP registered for testing previously and the JenkinsJSExtensionsTest has been updated to pass.

**Submitter checklist**
- [x] Change is code complete and matches issue description
- [x] Apppropriate unit or acceptance tests or explaination to why this change has no tests

**Reviewer checklist**
- [ ] Run the changes and verified the change matches the issue description
- [x] Reviewed the code
- [x] Verified that the appropriate tests have been written or valid explaination given

@reviewbybees

### CODE REVIEW

File: blueocean-dashboard/src/main/js/components/PullRequest.jsx [hunk 1]
```diff
@@ -52,7 +53,10 @@ export default class PullRequest extends Component {
             <td>{title || '-'}</td>
             <td>{author || '-'}</td>
             <td><ReadableDate date={endTime} liveUpdate /></td>
-            <td><RunPipeline organization={organization} pipeline={pipelineName} branch={name} /></td>
+            <td>
+                <RunPipeline organization={organization} pipeline={pipelineName} branch={name} />
+                <Extensions.Renderer extensionPoint="jenkins.pipeline.pullrequests.list.action" />
```
Comment (scherler) [inline blueocean-dashboard/src/main/js/components/PullRequest.jsx]: why did you not name it jenkins.dashboard but jenkins.pipeline?

Comment (cliffmeyers) [inline blueocean-dashboard/src/main/js/components/PullRequest.jsx]: I was trying to be consistent with some of the other extension point names:
* jenkins.pipeline.list.top   (area above the main pipelines list)
* jenkins.pipeline.run.result   (from run details)

The other extension point we currently have in "jenkins.test.result" from the test page.

Comment (scherler) [inline blueocean-dashboard/src/main/js/components/PullRequest.jsx]: yeah got ya, I thought we had renamed those to dashboard.

File: blueocean-dashboard/src/main/js/components/PipelineRowItem.jsx [hunk 2]
```diff
@@ -87,7 +87,7 @@ export default class PipelineRowItem extends Component {
                 <td>{pullRequestsLink}</td>
                 <td>
                     <Favorite />
-                    <Extensions.Renderer extensionPoint="jenkins.dashboard.item.action" />
+                    <Extensions.Renderer extensionPoint="jenkins.pipeline.list.action" />
```
Comment (kzantow) [inline blueocean-dashboard/src/main/js/components/PipelineRowItem.jsx]: 👍

File: blueocean-plugin/src/test/java/io/jenkins/blueocean/jsextensions/JenkinsJSExtensionsTest.java [hunk 3]
```diff
@@ -56,7 +56,7 @@ public void test() {
             } else if ("blueocean-personalization".equals(pluginId)) {
                 Assert.assertEquals(1, extensionPoints.size());
                 Assert.assertEquals("components/ActionLink", extensionPoints.get(0).get("component"));
-                Assert.assertEquals("jenkins.dashboard.item.action", extensionPoints.get(0).get("extensionPoint"));
+                Assert.assertEquals("jenkins.pipeline.list.action", extensionPoints.get(0).get("extensionPoint"));
```
Comment (kzantow) [inline blueocean-plugin/src/test/java/io/jenkins/blueocean/jsextensions/JenkinsJSExtensionsTest.java]: only 🐜  I have about these extension points (and the existing ones) is `jenkins.` is pretty unnecessary, tbh... we might want to revisit this before a non-alpha release

**Comment by kzantow** [comment]
:bee: LGTM from the code perspective, will run + verify shortly

File: blueocean-plugin/src/test/java/io/jenkins/blueocean/jsextensions/JenkinsJSExtensionsTest.java (hunk 3 above)
Comment (michaelneale) [inline blueocean-plugin/src/test/java/io/jenkins/blueocean/jsextensions/JenkinsJSExtensionsTest.java]: fair point

**Comment by cliffmeyers** [comment]
@kzantow this one required a merge of master, should be ready now.

**Comment by i386** [comment]
@imeredith this should unblock you now

**Pull request merged**